packages/sdk-keysigner/index.ts

```diff
@@ -78,6 +78,62 @@ export class BrowserKeySigner implements CredentialSigner<KeyAssertion> {
     ])
   }
 
+  toHex = (buffer: ArrayBuffer): string => {
+    const view = new Uint8Array(buffer)
+    let hexString = ""
+    for (const byte of view) {
+      const hexByte = byte.toString(16)
+      hexString += hexByte.padStart(2, "0")
+    }
+    return hexString.toLowerCase()
+  }
+
+  exportPublicKeyInPemFormat = async (key: CryptoKeyPair): Promise<string> => {
+    const exported = await crypto.subtle.exportKey('spki', key.publicKey)
+    const pem = `-----BEGIN PUBLIC KEY-----\n${toBase64(exported)}\n-----END PUBLIC KEY-----`
+    return pem
+  }
+
+  async create(challenge: UserRegistrationChallenge): Promise<KeyAttestation> {
+    const publicKeyPem = await this.exportPublicKeyInPemFormat(this.options.keyPair)
+    const clientData = JSON.stringify({
+        type: 'key.create',
+        challenge: challenge.challenge,
+        origin: this.options.appOrigin,
+        crossOrigin: this.options.crossOrigin ?? false,
+      })
+    const clientDataHash = await crypto.subtle.digest('SHA-256', new TextEncoder().encode(clientData))
+    const clientDataHashHex = this.toHex(clientDataHash)
+    const credInfoFingerprint = JSON.stringify({
+      clientDataHash: clientDataHashHex,
+      publicKey: publicKeyPem,
+    })
+    let rawSignature: ArrayBuffer
+    const algorithm = this.options.keyPair.privateKey.algorithm.name
+    if (algorithm == 'ECDSA') {
+      rawSignature = await crypto.subtle.sign(
+        { name: 'ECDSA', hash: { name: 'SHA-256' } },
+        this.options.keyPair.privateKey,
+        new TextEncoder().encode(credInfoFingerprint)
+      )
+    } else {
+      throw new Error(`${algorithm} is not supported`)
+    }
+    const signature = this.rawSignatureToAns1(new Uint8Array(rawSignature))
+    const attestationData = JSON.stringify({
+      publicKey: publicKeyPem,
+      signature: this.toHex(signature) 
+    })
+
+    return {
+      credentialKind: 'Key',
+      credentialInfo: {
+        credId: this.options.credId,
```

Comment:
Yeah I don't have a strong opinion about this. In my testing example I generate a random credId and pass it to the constructor. I thought we could let the customer choose a credId. But since it is possible to pass a name to the credential that does not make much sense. I'll create a random one.

Reply:
If credId is not passed to the constructor it is randomly generated.

Reply:
thinking more about this, maybe we do want them to pick the `credId`, or have a way to generate a "stable" `credId`. one problem I'm thinking generating a random one is, if they can the `create` multiple times, they can potentially create the credential over and over.

and they have no way of knowing the right `credId` to configure the sign with in their code if they are not the ones choosing. that's another usability problem I think.

Reply:
> anyway, sorry I think what you had before is correct. they must provide a credId

Yes so with my approach either they don't provide a credId and it randomly generated or they provide one and we use it. If it is randomly generated they can get it from the returned object `credentialInfo.credId`. 
Are you with this approach?

> credId is base64url encoded, we should ensure it's base64url encoded here too to be consistent I think

I looked into that and it is actually not trivial to make sure a string is base64 or not. All the ascii string once padded can actually be base64 decoded. I am afraid the check can actually introduce some uncontrolled edge cases.

Reply:
yeah maybe we base64 it when we generate one, otherwise just leave it as is. 👍